platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/cli/FormatSwirldsLogCommand.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.cli;
+
+import com.swirlds.cli.commands.SwirldsLogCommand;
+import com.swirlds.cli.logging.LogProcessor;
+import com.swirlds.cli.utility.AbstractCommand;
+import com.swirlds.cli.utility.SubcommandOf;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Objects;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+        name = "format",
+        mixinStandardHelpOptions = true,
+        description = "Generate an html formatted version of swirlds log files.")
+@SubcommandOf(SwirldsLogCommand.class)
+public class FormatSwirldsLogCommand extends AbstractCommand {
+
+    /**
+     * The path to the root directory where the input log files are contained.
+     */
+    private Path inputDirectory;
+
+    /**
+     * The path where the output file will be saved.
+     */
+    private Path outputPath;
+
+    @CommandLine.Option(
+            names = {"-i", "--input"},
+            description = "The path to the root directory where the input log files are contained."
+                    + "Defaults to the current working directory")
+    private void setInputDirectory(@NonNull final Path inputDirectory) {
+        if (!Files.isDirectory(inputDirectory)) {
+            throw new IllegalArgumentException(
+                    "Input directory '%s' does not exist, or is not a directory".formatted(inputDirectory));
+        }
+
+        this.inputDirectory = Objects.requireNonNull(inputDirectory);
+    }
```

Comment:
It might be useful if there was a way to specify multiple files in different directories. For JRS, the log files are not stored in the GCP bucket all in one directory.

Reply:
The input directory is used as a root, and the log processor delves into child directories with the layout of the JRS data folder, which I think should suffice for the initial implementation. My thought was that we could make this more versatile if necessary in the future.

Thoughts?